### ISSUE TITLE
[Feature request] Modularization Enhancement for Document Scanner API Library

### ISSUE THREAD

**Issue by aerocube-tech** [googlesamples/mlkit#816]
**What's your feature request? Please describe.**
The current Document Scanner API Library provides comprehensive functionality for document scanning, editing, and image manipulation. However, there's a growing need to modularize these features, allowing developers to integrate specific functionalities seamlessly into their applications. This feature request aims to enhance the library by breaking down its components into modular units, enabling developers to pick and choose functionalities according to their requirements.

Modularization Goals:

Scanning Module: Create a standalone module focused solely on document scanning capabilities. This module should provide APIs to capture images, detect edges, and enhance document readability.
Editing Module: Introduce a modular editing feature that encompasses functionalities like rotate, crop, filter, and adjust. This module should offer granular control over image manipulation, allowing developers to integrate specific editing features as needed.
Gallery Integration: Implement a separate module dedicated to gallery integration, enabling developers to seamlessly retrieve images from the device's gallery for further processing.
Benefits:

Enhanced Flexibility: By modularizing the library, developers can select and integrate only the functionalities they require, reducing unnecessary overhead and optimizing application performance.
Simplified Integration: Modular units provide a clear and concise interface, making it easier for developers to integrate specific features into their applications without dealing with complex dependencies.
Scalability: The modular approach lays the groundwork for future enhancements and additions to the library, ensuring scalability and adaptability to evolving developer needs and industry standards.
Implementation Strategy:

Component Decoupling: Identify and decouple existing functionalities into separate modules, ensuring minimal dependencies between components.
API Design: Design clear and intuitive APIs for each modular unit, adhering to established design principles and developer expectations.
Documentation and Examples: Provide comprehensive documentation and usage examples for each module, facilitating seamless integration and adoption by developers.

**Mobile environment**
Android

**Comment by mubaarakhassan** [comment]
Yes, would love to see the document scanning capabilities, such as rectangle/edge detection, exposed as a standalone API. Very similar to what Apple offers via the Vision Framework. On iOS you are able to use the rectangle detection seamlessly within your own camera flow, without needing to launch a separate UI activity.
